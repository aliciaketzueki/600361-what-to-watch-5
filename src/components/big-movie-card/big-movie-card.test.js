import React from "react";
import renderer from "react-test-renderer";
import BigMovieCard from "./big-movie-card";
import {Router as BrowserRouter} from "react-router-dom";
import {Provider} from "react-redux";
import browserHistory from "../../browser-history";
import {store} from "../../index";

const film = {
  id: 1,
  name: `The Grand Budapest Hotel`,
  posterImage: `img/the-grand-budapest-hotel-poster.jpg`,
  previewImage: `img/the-grand-budapest-hotel.jpg`,
  backgroundImage: `img/the-grand-budapest-hotel-bg.jpg`,
  backgroundColor: `#ffffff`,
  videoLink: `https://some-link`,
  previewVideoLink: `https://some-link`,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
  rating: 8.9,
  scoresCount: 240,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
  runTime: 99,
  genre: `Comedy`,
  released: 2014,
  isFavorite: false
};

const review = {
  id: 1,
  user: {
    id: 4,
    name: "Kate Muir"
  },
  rating: 8.9,
  comment: "Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.",
  date: "2019-05-08T14:13:56.569Z"
};

const reviews = [review];

const isFull = {
  with: true,
  without: false
}

describe(`Render BigMovieCard`, () => {
  test.each([
    [`with`, true],
    [`without`, false],
  ])(`%s full content`, (_expected, isFull) => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <BrowserRouter history={browserHistory}>
              <BigMovieCard
                isFull={isFull}
                isReview={true}
                film={film}
                reviews={reviews}
              />
            </BrowserRouter>
          </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test.each([
    [`with`, true],
    [`without`, false],
  ])(`%s form review`, (_expected, isReview) => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <BrowserRouter history={browserHistory}>
              <BigMovieCard
                isFull={true}
                isReview={isReview}
                film={film}
                reviews={reviews}
              />
            </BrowserRouter>
          </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
