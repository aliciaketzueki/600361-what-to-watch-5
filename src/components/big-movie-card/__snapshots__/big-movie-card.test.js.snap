// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render BigMovieCard with form review 1`] = `
<section
  className="movie-card movie-card--full"
>
  <div
    className="movie-card__header"
  >
    <MovieCardHead
      film={
        Object {
          "backgroundColor": "#ffffff",
          "backgroundImage": "img/the-grand-budapest-hotel-bg.jpg",
          "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
          "director": "Wes Andreson",
          "genre": "Comedy",
          "id": 1,
          "isFavorite": false,
          "name": "The Grand Budapest Hotel",
          "posterImage": "img/the-grand-budapest-hotel-poster.jpg",
          "previewImage": "img/the-grand-budapest-hotel.jpg",
          "previewVideoLink": "https://some-link",
          "rating": 8.9,
          "released": 2014,
          "runTime": 99,
          "scoresCount": 240,
          "starring": Array [
            "Bill Murray",
            "Edward Norton",
            "Jude Law",
            "Willem Dafoe",
            "Saoirse Ronan",
          ],
          "videoLink": "https://some-link",
        }
      }
      headerNav={true}
    />
    <MovieCardPoster
      classname="small"
      film={
        Object {
          "backgroundColor": "#ffffff",
          "backgroundImage": "img/the-grand-budapest-hotel-bg.jpg",
          "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
          "director": "Wes Andreson",
          "genre": "Comedy",
          "id": 1,
          "isFavorite": false,
          "name": "The Grand Budapest Hotel",
          "posterImage": "img/the-grand-budapest-hotel-poster.jpg",
          "previewImage": "img/the-grand-budapest-hotel.jpg",
          "previewVideoLink": "https://some-link",
          "rating": 8.9,
          "released": 2014,
          "runTime": 99,
          "scoresCount": 240,
          "starring": Array [
            "Bill Murray",
            "Edward Norton",
            "Jude Law",
            "Willem Dafoe",
            "Saoirse Ronan",
          ],
          "videoLink": "https://some-link",
        }
      }
    />
  </div>
  <div
    className="add-review"
  >
    <FormAddReview
      checkValid={[Function]}
      commentValid={true}
      email=""
      emailValid={true}
      filmId={1}
      formErrors={
        Object {
          "email": "",
          "password": "",
          "textReview": "",
        }
      }
      formValid={false}
      handleFieldChange={[Function]}
      handleSubmit={[Function]}
      password=""
      passwordValid={true}
      rating="3"
      textReview=""
    />
  </div>
</section>
`;

exports[`Render BigMovieCard with full content 1`] = `
<section
  className="movie-card movie-card--full"
>
  <div
    className="movie-card__header"
  >
    <MovieCardHead
      film={
        Object {
          "backgroundColor": "#ffffff",
          "backgroundImage": "img/the-grand-budapest-hotel-bg.jpg",
          "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
          "director": "Wes Andreson",
          "genre": "Comedy",
          "id": 1,
          "isFavorite": false,
          "name": "The Grand Budapest Hotel",
          "posterImage": "img/the-grand-budapest-hotel-poster.jpg",
          "previewImage": "img/the-grand-budapest-hotel.jpg",
          "previewVideoLink": "https://some-link",
          "rating": 8.9,
          "released": 2014,
          "runTime": 99,
          "scoresCount": 240,
          "starring": Array [
            "Bill Murray",
            "Edward Norton",
            "Jude Law",
            "Willem Dafoe",
            "Saoirse Ronan",
          ],
          "videoLink": "https://some-link",
        }
      }
      headerNav={true}
    />
    <MovieCardPoster
      classname="small"
      film={
        Object {
          "backgroundColor": "#ffffff",
          "backgroundImage": "img/the-grand-budapest-hotel-bg.jpg",
          "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
          "director": "Wes Andreson",
          "genre": "Comedy",
          "id": 1,
          "isFavorite": false,
          "name": "The Grand Budapest Hotel",
          "posterImage": "img/the-grand-budapest-hotel-poster.jpg",
          "previewImage": "img/the-grand-budapest-hotel.jpg",
          "previewVideoLink": "https://some-link",
          "rating": 8.9,
          "released": 2014,
          "runTime": 99,
          "scoresCount": 240,
          "starring": Array [
            "Bill Murray",
            "Edward Norton",
            "Jude Law",
            "Willem Dafoe",
            "Saoirse Ronan",
          ],
          "videoLink": "https://some-link",
        }
      }
    />
  </div>
  <div
    className="add-review"
  >
    <FormAddReview
      checkValid={[Function]}
      commentValid={true}
      email=""
      emailValid={true}
      filmId={1}
      formErrors={
        Object {
          "email": "",
          "password": "",
          "textReview": "",
        }
      }
      formValid={false}
      handleFieldChange={[Function]}
      handleSubmit={[Function]}
      password=""
      passwordValid={true}
      rating="3"
      textReview=""
    />
  </div>
</section>
`;

exports[`Render BigMovieCard without form review 1`] = `
<section
  className="movie-card movie-card--full"
>
  <div
    className="movie-card__hero"
  >
    <MovieCardHead
      film={
        Object {
          "backgroundColor": "#ffffff",
          "backgroundImage": "img/the-grand-budapest-hotel-bg.jpg",
          "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
          "director": "Wes Andreson",
          "genre": "Comedy",
          "id": 1,
          "isFavorite": false,
          "name": "The Grand Budapest Hotel",
          "posterImage": "img/the-grand-budapest-hotel-poster.jpg",
          "previewImage": "img/the-grand-budapest-hotel.jpg",
          "previewVideoLink": "https://some-link",
          "rating": 8.9,
          "released": 2014,
          "runTime": 99,
          "scoresCount": 240,
          "starring": Array [
            "Bill Murray",
            "Edward Norton",
            "Jude Law",
            "Willem Dafoe",
            "Saoirse Ronan",
          ],
          "videoLink": "https://some-link",
        }
      }
    />
    <div
      className="movie-card__wrap"
    >
      <MovieCardDescr
        film={
          Object {
            "backgroundColor": "#ffffff",
            "backgroundImage": "img/the-grand-budapest-hotel-bg.jpg",
            "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
            "director": "Wes Andreson",
            "genre": "Comedy",
            "id": 1,
            "isFavorite": false,
            "name": "The Grand Budapest Hotel",
            "posterImage": "img/the-grand-budapest-hotel-poster.jpg",
            "previewImage": "img/the-grand-budapest-hotel.jpg",
            "previewVideoLink": "https://some-link",
            "rating": 8.9,
            "released": 2014,
            "runTime": 99,
            "scoresCount": 240,
            "starring": Array [
              "Bill Murray",
              "Edward Norton",
              "Jude Law",
              "Willem Dafoe",
              "Saoirse Ronan",
            ],
            "videoLink": "https://some-link",
          }
        }
      />
    </div>
  </div>
  <div
    className="movie-card__wrap movie-card__translate-top"
  >
    <div
      className="movie-card__info"
    >
      <MovieCardPoster
        classname="big"
        film={
          Object {
            "backgroundColor": "#ffffff",
            "backgroundImage": "img/the-grand-budapest-hotel-bg.jpg",
            "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
            "director": "Wes Andreson",
            "genre": "Comedy",
            "id": 1,
            "isFavorite": false,
            "name": "The Grand Budapest Hotel",
            "posterImage": "img/the-grand-budapest-hotel-poster.jpg",
            "previewImage": "img/the-grand-budapest-hotel.jpg",
            "previewVideoLink": "https://some-link",
            "rating": 8.9,
            "released": 2014,
            "runTime": 99,
            "scoresCount": 240,
            "starring": Array [
              "Bill Murray",
              "Edward Norton",
              "Jude Law",
              "Willem Dafoe",
              "Saoirse Ronan",
            ],
            "videoLink": "https://some-link",
          }
        }
      />
      <div
        className="movie-card__desc"
      >
        <nav
          className="movie-nav movie-card__nav"
        >
          <ul
            className="movie-nav__list"
          >
            <li
              className="movie-nav__item movie-nav__item--active"
            >
              <a
                aria-current="page"
                className="movie-nav__link active"
                href="/"
                onClick={[Function]}
                style={Object {}}
              >
                Overview
              </a>
            </li>
            <li
              className="movie-nav__item "
            >
              <a
                aria-current="page"
                className="movie-nav__link active"
                href="/"
                onClick={[Function]}
                style={Object {}}
              >
                Details
              </a>
            </li>
            <li
              className="movie-nav__item "
            >
              <a
                aria-current="page"
                className="movie-nav__link active"
                href="/"
                onClick={[Function]}
                style={Object {}}
              >
                Reviews
              </a>
            </li>
          </ul>
        </nav>
        <div
          className="movie-rating"
        >
          <div
            className="movie-rating__score"
          >
            8.9
          </div>
          <p
            className="movie-rating__meta"
          >
            <span
              className="movie-rating__level"
            >
              Very good
            </span>
            <span
              className="movie-rating__count"
            >
              240
               ratings
            </span>
          </p>
        </div>
        <div
          className="movie-card__text"
        >
          <p>
            In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
          </p>
          <p
            className="movie-card__director"
          >
            <strong>
              Director: 
              Wes Andreson
            </strong>
          </p>
          <p
            className="movie-card__starring"
          >
            <strong>
              Starring: 
              Bill Murray, Edward Norton, Jude Law, Willem Dafoe
               and other
            </strong>
          </p>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`Render BigMovieCard without full content 1`] = `
<section
  className="movie-card "
>
  <MovieCardHead
    film={
      Object {
        "backgroundColor": "#ffffff",
        "backgroundImage": "img/the-grand-budapest-hotel-bg.jpg",
        "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
        "director": "Wes Andreson",
        "genre": "Comedy",
        "id": 1,
        "isFavorite": false,
        "name": "The Grand Budapest Hotel",
        "posterImage": "img/the-grand-budapest-hotel-poster.jpg",
        "previewImage": "img/the-grand-budapest-hotel.jpg",
        "previewVideoLink": "https://some-link",
        "rating": 8.9,
        "released": 2014,
        "runTime": 99,
        "scoresCount": 240,
        "starring": Array [
          "Bill Murray",
          "Edward Norton",
          "Jude Law",
          "Willem Dafoe",
          "Saoirse Ronan",
        ],
        "videoLink": "https://some-link",
      }
    }
  />
  <div
    className="movie-card__wrap"
  >
    <div
      className="movie-card__info"
    >
      <MovieCardPoster
        film={
          Object {
            "backgroundColor": "#ffffff",
            "backgroundImage": "img/the-grand-budapest-hotel-bg.jpg",
            "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
            "director": "Wes Andreson",
            "genre": "Comedy",
            "id": 1,
            "isFavorite": false,
            "name": "The Grand Budapest Hotel",
            "posterImage": "img/the-grand-budapest-hotel-poster.jpg",
            "previewImage": "img/the-grand-budapest-hotel.jpg",
            "previewVideoLink": "https://some-link",
            "rating": 8.9,
            "released": 2014,
            "runTime": 99,
            "scoresCount": 240,
            "starring": Array [
              "Bill Murray",
              "Edward Norton",
              "Jude Law",
              "Willem Dafoe",
              "Saoirse Ronan",
            ],
            "videoLink": "https://some-link",
          }
        }
      />
      <MovieCardDescr
        film={
          Object {
            "backgroundColor": "#ffffff",
            "backgroundImage": "img/the-grand-budapest-hotel-bg.jpg",
            "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
            "director": "Wes Andreson",
            "genre": "Comedy",
            "id": 1,
            "isFavorite": false,
            "name": "The Grand Budapest Hotel",
            "posterImage": "img/the-grand-budapest-hotel-poster.jpg",
            "previewImage": "img/the-grand-budapest-hotel.jpg",
            "previewVideoLink": "https://some-link",
            "rating": 8.9,
            "released": 2014,
            "runTime": 99,
            "scoresCount": 240,
            "starring": Array [
              "Bill Murray",
              "Edward Norton",
              "Jude Law",
              "Willem Dafoe",
              "Saoirse Ronan",
            ],
            "videoLink": "https://some-link",
          }
        }
      />
    </div>
  </div>
</section>
`;
